src/screens/map-screen.tsx

```diff
@@ -1,195 +1,29 @@
-import { useState, useEffect, useRef } from "react";
+import { useEffect } from "react";
 import { View, StyleSheet } from "react-native";
-import { LatLng } from "react-native-maps";
 import * as Location from "expo-location";
-import * as Cellular from "expo-cellular";
-import { LocationObject } from "expo-location";
+
 import AppHeader from "../components/app-header";
 import MapViewContainer from "../components/map-view-container";
 import RouteButtonContainer from "../components/route-button-container";
 import InfoContainer from "../components/info-container";
-import useIdentifyUser from "../hooks/use-identify-user";
-import useRoutes from "../hooks/use-routes";
 import NavigatorTab from "../components/navigator-tab";
+import { useTypedSelector } from "../store";
 
 const MapScreen = () => {
-	const [, setErrorMsg] = useState<string | null>(null);
-	const [curLocation, setCurLocation] = useState<LocationObject | null>(null);
-	const [mobileNetCode, setMobileNetCode] = useState<string | null>(null);
-	const [routeCoordinates, setRouteCoordinates] = useState<Array<LatLng>>([]);
-	const [tempWaypoints, setTempWaypoints] = useState<
-		Array<null | {
-			routeId: string;
-			location: Location.LocationObject;
-			mobileNetCode: string;
-		}>
-	>([]);
-	const [showRoute, setShowRoute] = useState<boolean>(true);
-	const [trackingInterval] = useState<number>(2500);
-	const [sendingInterval] = useState<number>(15000);
-	const [isTracking, setIsTracking] = useState<boolean>(false);
-	const [userId, setUserId] = useState<string | null>(null);
-	const [routeId, setRouteId] = useState<string | null>(null);
-	const waypointUpdateRef = useRef<() => void>();
-	const waypointSendRef = useRef<() => void>();
-	const identifyUser = useIdentifyUser();
-	const { startRoute, sendWaypoint, deactivateRoute } = useRoutes();
+	const userId = useTypedSelector((state) => state.user.userId);
 
-	/**
-	 * Requests permissions to use device location.
-	 * Gets device location and sets current location state if permissions is allowed.
-	 * Otherwise sets error message state which is not currently used anywhere.
-	 * Also gets cellular network operators MNC (Mobile Network Code), identifies user and sets user ID state.
-	 * MNC code is a null if SIM card is not at the device or there is no cellular service available.
-	 */
 	useEffect(() => {
 		(async () => {
-			const { status } = await Location.requestForegroundPermissionsAsync();
-			if (status !== "granted") {
-				setErrorMsg("Permission to access location was denied");
-				return;
-			}
-			const location = await Location.getCurrentPositionAsync({});
-			setCurLocation(location);
-			const networkCode = await Cellular.getMobileNetworkCodeAsync();
-			setMobileNetCode(networkCode);
-			const uid = await identifyUser();
-			setUserId(uid);
+			await Location.requestForegroundPermissionsAsync();
```

Comment:
Does this have to be an IIFE? The result of the promise isn't used so this should be equivalent to calling `Location.requestForegroundPermissionsAsync` without `await`, right?

Reply:
Yeah I guess so. I did remove async / await for now. This used to have incomplete error handling previously, but I did remove it as it wasn't used for anything. Perhaps we may implement that later as this application has not much to use if the location is not allowed...